### PR TITLE
ci: enable clang based RBE in Azure pipelines

### PR DESCRIPTION
**Pull request by lizan**
Signed-off-by: Lizan Zhou <lizan@tetrate.io>

Description:
Minimum configuration to enable RBE and run release pipeline in Azure pipelines.

Risk Level: Low
Testing:
Docs Changes: N/A
Release Notes: N/A

### CODE REVIEW

File: bazel/toolchains/clang/BUILD [hunk 1]
```diff
@@ -0,0 +1,169 @@
+licenses(["notice"])  # Apache 2
+
+# THIS FILE IS GENERATED, DO NOT EDIT.
+# TODO(lizan): Document how to generate this file
```
Comment (jmarantz) [inline bazel/toolchains/clang/BUILD]: will this be done before you mark this ready for review?

Comment (lizan) [reply bazel/toolchains/clang/BUILD]: Yes, I will add this once CI is happy.

**Comment by lizan** [comment]
@jmarantz this is ready to review, I will try to add more docs how to use RBE as follow up.

File: bazel/toolchains/configs/clang/bazel_0.28.0/cc/BUILD [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+        "strip": "/usr/bin/strip"},
+    compile_flags = ["-U_FORTIFY_SOURCE",
+    "-fstack-protector",
+    "-Wall",
+    "-Wthread-safety",
+    "-Wself-assign",
+    "-fcolor-diagnostics",
+    "-fno-omit-frame-pointer"],
+    opt_compile_flags = ["-g0",
+    "-O2",
```
Comment (jmarantz) [inline bazel/toolchains/configs/clang/bazel_0.28.0/cc/BUILD]: nit: indentation

File: bazel/toolchains/configs/clang/bazel_0.28.0/cc/BUILD [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    "-D_FORTIFY_SOURCE=1",
+    "-DNDEBUG",
+    "-ffunction-sections",
+    "-fdata-sections"],
+    dbg_compile_flags = ["-g"],
+    cxx_flags = ["-std=c++0x"],
+    link_flags = ["-fuse-ld=gold",
+    "-Wl,-no-as-needed",
```
Comment (jmarantz) [inline bazel/toolchains/configs/clang/bazel_0.28.0/cc/BUILD]: nit: indentation

File: bazel/toolchains/configs/gcc/bazel_0.28.0/cc/cc_wrapper.sh [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+#
+# Ship the environment to the C++ action
+#
+set -eu
+
+# Set-up the environment
+
+
+# Call the C++ compiler
+/usr/bin/gcc "$@"
```
Comment (jmarantz) [inline bazel/toolchains/configs/gcc/bazel_0.28.0/cc/cc_wrapper.sh]: exec?

**Comment by lizan** [comment]
@jmarantz sorry if it wasn't clear in README.md, all files in `bazel/toolchains/configs` are generated without manual edit.

**Comment by jmarantz** [comment]
OK ignore my 'indentation' nits then :)

**Comment by lizan** [comment]
and `cc_wrapper` as well, they are not actually used but cc_config generates them, even in your local run. I'd leave them as it is as. :)

**Comment by jmarantz** [comment]
alright then...I probably should've read the README more carefully

**Comment by lizan** [comment]
Merging this to roll out RBE runs on master. Azure pipelines are not marked required so it should not block PRs. Let's iterate from here.

**Pull request merged**